draft-ietf-quic-transport.md

```diff
@@ -2205,16 +2205,29 @@ connection.
 The endpoint MUST use unpredictable data in every PATH_CHALLENGE frame so that
 it can associate the peer's response with the corresponding PATH_CHALLENGE.
 
+An endpoint MUST expand datagrams that contain a PATH_CHALLENGE frame to at
+least the smallest allowed maximum datagram size of 1200 bytes.  Sending UDP
+datagrams of this size ensures that the network path from the endpoint to the
+peer can be used for QUIC; see {{datagram-size}}.
+
 
 ### Path Validation Responses
 
 On receiving a PATH_CHALLENGE frame, an endpoint MUST respond by echoing the
-data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame. A
-PATH_RESPONSE frame does not need to be sent on the network path where the
-PATH_CHALLENGE was received; a PATH_RESPONSE can be sent on any network path.
-An endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE
+data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame.  An
+endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE
 frame unless constrained by congestion control.
 
+A PATH_RESPONSE frame SHOULD be sent on the network path where the
+PATH_CHALLENGE was received.  This ensures the path is functional in both
+directions.  This requirement MUST NOT be enforced by the endpoint that
+initiates path validation as that would enable an attack on migration; see
+{{off-path-forward}}.
+
+An endpoint SHOULD expand datagrams that contain a PATH_RESPONSE frame to at
```

Comment:
Why is this optional?
```suggestion
An endpoint MUST expand datagrams that contain a PATH_RESPONSE frame to at
```

Reply:
I requested it be a SHOULD for two reasons:
 1) It's unenforceable, and unenforceable MUSTs are unfortunate.
 2) If we ever did multipath, it may make sense to send PATH_RESPONSE on a different path than the PATH_CHALLENGE, and a SHOULD makes that possible.

Reply:
I was happy to accommodate Ian's request, but I think that MUST is slightly better, even if it is unenforceable (I'm sure that Marten will be happy to enforce it in practice.)

Reply:
A SHOULD suggests there are potentially good reasons where an implementation might deviate from the recommendation. If there aren't any, then it's a MUST. It doesn't have to be enforceable, since not doing it means the protocol might not work as intended. That is the case here -- not following the MUST means that the protocol might not work.

In terms of the value in deviating from this recommendation, I don't see this degree of freedom being useful for multipath, though it's likely I'm not seeing what you're thinking. That said, I don't think we should design multipath here.